### DISCUSSION TITLE
Running via EC2 instance role to capture terraform changes in AWS?

### DISCUSSION THREAD

**Discussion by JayeMcC** [iann0036/iamlive#79] [Q&A] [answered]
I have a remote dev setup on an ec2 instance that relies on an instance role for authentication. I assume this is either unsupported or would require a different setup due to not having a .aws directory to pull authentication details from.

For context, I want to run Terraform to create AWS infrastructure.

**Comment by iann0036** [comment, marked as answer]
Hi @JayeMcC,

iamlive uses the default credential chain, so you should be able to use the instance role out-of-the-box. You'll just need to make sure you set the required environment variables just prior to executing Terraform (in the same terminal session as Terraform).

In proxy mode, the required environment variables are:

```
export HTTP_PROXY=http://127.0.0.1:10080
export HTTPS_PROXY=http://127.0.0.1:10080
export AWS_CA_BUNDLE=~/.iamlive/ca.pem
```

or in CSM mode, the required environment variables are:

```
export AWS_CSM_ENABLED=true
export AWS_CSM_PORT=31000
export AWS_CSM_HOST=127.0.0.1
```
